src/main/java/uk/gov/register/db/ItemQueryDAO.java

```diff
@@ -21,11 +21,14 @@
     @RegisterMapper(ItemMapper.class)
     Optional<Item> getItemBySHA256(@Bind("sha256hex") String sha256Hash, @Define("schema") String schema );
 
-    //Note: This is fine for small data registers like country
     @SqlQuery("select sha256hex, content from \"<schema>\".item order by sha256hex")
     @RegisterMapper(ItemMapper.class)
     Collection<Item> getAllItemsNoPagination( @Define("schema") String schema );
 
+    @SqlQuery("select sha256hex, content from \"<schema>\".item i where exists(select 1 from \"<schema>\".<entry_table> e where i.sha256hex = e.sha256hex) order by sha256hex")
```

Comment:
given that the `sha256hex` is not a human meaningful string is there any reason to sort by it? What would an alternative sort option be?

Reply:
The only reason to use it is that we need some kind of ordering of the blobs in order to split them over pages.

Although I agree that the hash doesn't make a lot of sense. If blobs get added to the register while you are paging through it you will get strange results.

Actually I just noticed today that there is an ADR for this here: https://github.com/openregister/openregister-java/blob/master/doc/arch/adr-008-items-resource.md

This suggests ordering items by insertion order, which would mean adding a serial field to the items table, but seems like a better approach than what I've done here. Would it be ok to comment this endpoint out for now and address this in a future PR? It already needs further work to add the pagination.

Reply:
yes, that is fine, can we make a trello card for it so we don't forget?

Reply:
Turns out there was already a card, https://trello.com/c/FiMDw8PF/689-implement-blobs-endpoint-in-orj - but it's not actually in scope for this quarter at the moment.

I would like to get it done before then if we have time though, because it makes much easier to fetch all entries and blobs for a register.